db/batchlog_manager.cc

```diff
@@ -123,7 +123,9 @@ future<> db::batchlog_manager::start() {
             (void)do_batch_log_replay().handle_exception([] (auto ep) {
```

Comment:
I think there's still a problem here. do_batch_log_replay doesn't enter the gate, so it won't synchronize with stop().

Reply:
That's exactly the reason I added `_sem.wait()` to stop() in https://github.com/scylladb/scylla/pull/9010/commits/19e026ca3b5e94eaa755f7a064f97a5b0f84ee19

Reply:
@avikivity ping, please re-review.